framework/encode/custom_vulkan_struct_encoders.cpp

```diff
@@ -90,6 +90,11 @@ void EncodeStruct(ParameterEncoder* encoder, const VkDeviceOrHostAddressConstKHR
     encoder->EncodeVkDeviceAddressValue(value.deviceAddress);
 }
 
+void EncodeStruct(ParameterEncoder* encoder, const VkAccelerationStructureMotionInstanceDataNV& value)
+{
+    EncodeStruct(encoder, value.srtMotionInstance);
```

Comment:
I don't think this is portable. It relies on unpacking bit fields at encode time and repacking them at decode time (with the same exact alignment) to reconstruct the other union members.

What we do in similar cases is create a custom encode/decode handler for the *parent* structure (`VkAccelerationStructureMotionInstanceNV` in this case), and then explicitly encode/decode the union members based on the `type` value. Look at `Decoded_VkPerformanceValueINTEL` and `Decoded_VkAccelerationStructureGeometryKHR` for examples.

Reply:
Thanks. Fixed it.